### PR TITLE
Introduce OpenstackCredentialv3Convertor

### PR DESCRIPTION
**Pull request by MichalAugustyn**
Add support for openstack-cloud plugins's credentials
(OpenstackCredentialv3 class)

https://plugins.jenkins.io/openstack-cloud/

### CODE REVIEW

**Comment by MichalAugustyn** [comment]
Changes were tested locally on a microk8s cluster with:

> jenkins version:  2.204.2
> plugin versions: 2.47 (min) and 2.51 (latest)

There is a compilation error that I cannot resolve by myself.
`annotation @edu.umd.cs.findbugs.annotations.ReturnValuesAreNonnullByDefault is missing a default value for the element 'value'`

I have spent more than a day trying to resolve it and still failed. This error occurs after editing pom.xml even before introducing the OpenstackCredentialv3.java its Test class.

I really need your help since java is not my primary language and I am not so fluent in maven as well.

What I have also found is that this annotation is already deprecated.
http://findbugs.sourceforge.net/api/edu/umd/cs/findbugs/annotations/ReturnValuesAreNonnullByDefault.html

Except this single compilation error, all defined tests are passing. Credentials objects were also properly created in the Jenkins after I temporarily removed ReturnValuesAreNonnullByDefault annotation.

@jtnord please take a look at this PR. It would be great if you gave me a hint how to resolve the compilation problem at first.

File: docs/examples/openstack-credential-v3.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  name: "test-openstack-credential-v3"
+  labels:
+# so we know what type it is.
+    "jenkins.io/credentials-type": "openstackCredentialv3"
+  annotations:
+# description - can not be a label as spaces are not allowed
+    "jenkins.io/credentials-description" : "openstack credentials for you Jenkins in Kubernetes!"
+type: Opaque
+stringData:
+  userName: Y2FzdWFsTmFtZQ==               #casualName
```
Comment (jtnord) [inline docs/examples/openstack-credential-v3.yaml]: these shouldn't be base64 encoded as they are in a `stringData` not `string`

**Comment by jtnord** [comment]
> There is a compilation error that I cannot resolve by myself.
annotation @edu.umd.cs.findbugs.annotations.ReturnValuesAreNonnullByDefault is missing a default value for the element 'value'

> I have spent more than a day trying to resolve it and still failed. This error occurs after editing pom.xml even before introducing the OpenstackCredentialv3.java its Test class.

> I really need your help since java is not my primary language and I am not so fluent in maven as well.

> What I have also found is that this annotation is already deprecated.
http://findbugs.sourceforge.net/api/edu/umd/cs/findbugs/annotations/ReturnValuesAreNonnullByDefault.html

We should be using the code provided by [SpotBugs](https://github.com/spotbugs/spotbugs/blob/master/spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/ReturnValuesAreNonnullByDefault.java) which is not deprecated.  Something fishy is happening - I can not reproduce this locally using maven 3.5.2 or 3.6.2  (CI uses 3.5.4)

`ReturnValuesAreNonnullByDefault` does not even have a `value` parameter in either the spotbugs or findbugs version.

File: docs/examples/openstack-credential-v3.yaml (hunk 1 above)
Comment (MichalAugustyn) [reply docs/examples/openstack-credential-v3.yaml]: decrypted all records

**Comment by MichalAugustyn** [comment]
> We should be using the code provided by [SpotBugs](https://github.com/spotbugs/spotbugs/blob/master/spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/ReturnValuesAreNonnullByDefault.java) which is not deprecated. Something fishy is happening - I can not reproduce this locally using maven 3.5.2 or 3.6.2 (CI uses 3.5.4)
> 
> `ReturnValuesAreNonnullByDefault` does not even have a `value` parameter in either the spotbugs or findbugs version.

It does not seem to be related to the maven version. I have reproduced it on 3.6.2 and 3.6.3 maven versions.

```
git clone https://github.com/MichalAugustyn/kubernetes-credentials-provider-plugin.git -b augustyn/support-openstack plugin
cd $_
docker run -it --name maven-verify -v "$(pwd)":/usr/src/maven:z -w /usr/src/maven maven:3.6.3-jdk-8 mvn verify
```

I will consider asking SpotBugs' contributors for an advice.

**Comment by MichalAugustyn** [comment]
Found it!

Comparing full debug output gave me a hint that there might be a problem with one dependency.
Comparing dependency:tree gave the answer:
```
+[INFO] |  |  \- com.github.stephenc.findbugs:findbugs-annotations:jar:1.3.9-1:compile (optional) 
```

In this version, ReturnValuesAreNonnullByDefault contains following lines:
```
@Documented
@Nonnull
@TypeQualifierDefault(value = METHOD)
@Retention(value = RUNTIME)
public @interface ReturnValuesAreNonnullByDefault {
    When value();
}
```

I have excluded this library, especially it is marked as optional.

**Comment by MichalAugustyn** [comment]
hello @jtnord!

any plans to release a new version with this convertor ?:)

**Comment by jtnord** [comment]
sorry for the delay I dropped the ball here.  released as kubernetes-credentials-provider-0.14

**Pull request merged**